### ISSUE TITLE
Babel 5.6.7 spewing "Cannot read property key of null"

### ISSUE THREAD

**Issue by jsg2021** [babel/babel#1831]
[deleted red-herring]

Here is the stack trace:

```
Cannot read property 'key' of null
    at add (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/scope/index.js:317:19)
    at add (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/scope/index.js:299:11)
    at Scope.generateUidIdentifierBasedOnNode (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/scope/index.js:322:5)
    at NodePath.exit (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/transformation/transformers/es6/properties.computed.js:93:25)
    at NodePath.call (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/path/context.js:55:28)
    at NodePath.visit (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/path/context.js:106:12)
    at TraversalContext.visitSingle (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/context.js:106:12)
    at TraversalContext.visit (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/context.js:118:19)
    at Function.traverse.node (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/index.js:66:17)
    at NodePath.visit (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/path/context.js:105:26)
    at TraversalContext.visitMultiple (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/context.js:86:16)
    at TraversalContext.visit (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/context.js:116:19)
    at Function.traverse.node (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/index.js:66:17)
    at NodePath.visit (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/path/context.js:105:26)
    at TraversalContext.visitSingle (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/context.js:106:12)
    at TraversalContext.visit (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/context.js:118:19)
    at Function.traverse.node (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/index.js:66:17)
    at traverse (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/index.js:49:14)
    at PluginPass.transform (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/transformation/plugin-pass.js:38:32)
    at File.transform (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/transformation/file/index.js:590:12)
    at /Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/transformation/pipeline.js:125:19
    at File.wrap (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/transformation/file/index.js:604:16)
    at Pipeline.transform (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/transformation/pipeline.js:122:17)
    at transpile (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-loader/index.js:10:22)
    at Object.module.exports (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-loader/index.js:65:12)
```

**Comment by sebmck** [comment]
[I can't replicate this.](http://babeljs.io/repl/#?experimental=true&evaluate=true&loose=false&spec=false&code=const%20myCollisionFreeSudoProtectedMethod%20%3D%20'somename%20that%20cannot%20be%20easily%20typed'%3B%0A%0Aexport%20default%20%7B%0A%0A%20%20%20%20%5BmyCollisionFreeSudoProtectedMethod%5D%20()%20%7B%0A%20%20%20%20%20%20%20%20%2F%2Fetc...%0A%20%20%20%20%7D%2C%0A%0A%20%20%20%20mainMixinMethod%20()%20%7B%0A%0A%20%20%20%20%20%20%20%20this%5BmyCollisionFreeSudoProtectedMethod%5D()%3B%0A%0A%20%20%20%20%7D%0A%7D&playground=false) Can you provide a reproducible example in the REPL? What options are you using?

**Comment by jsg2021** [comment]
I'm using webpack and the babel loader... I think the computed keys were a red-herring.  I'm not sure how to make a sample or what's happening.

I do not have an exotic config...  webpack, and babel-loader... "?optional[]=runtime" ... I've tried without the optional... no effect.

I hand edited the `scope/index.js` and logged when it threw the error... the node.properties array has one to several `null` entries in it. Which, as the loop is written, are not expected.

I'm trying to narrow it down. I'm struggling on how to debug this.

**Comment by jsg2021** [comment]
I can say 5.6.6 does not exhibit the problem. :/

**Comment by sebmck** [comment]
Can you please put together a reproducible example?

**Comment by jsg2021** [comment]
I'm trying.

**Comment by jsg2021** [comment]
My project is very large.

**Comment by sebmck** [comment]
The beginning of the stack trace (before the error message) should have the filename of the file it errored on compiling. The entirety of that and the options used is all I'd need to reproduce.

**Comment by jsg2021** [comment]
here is one of the smaller files... as for options I don't have any... beyond the babel loader's runtime setting.  I don't have any plugins, nor rc file.

```
import React from 'react';
import {EventEmitter} from 'events';

function emit(o, event, ...data) {
	let e = o.state.dndEventEmitter;
	e.emit.apply(e, [event].concat(data));
}

const onDragStart = 'dnd:mixin:onDragStart';
const onDragEnd = 'dnd:mixin:onDragEnd';
const onDrag = 'dnd:mixin:onDrag';
const onDragOver = 'dnd:mixin:onDragOver';
const onDrop = 'dnd:mixin:onDrop';

export default {

	getInitialState () {
		return {
			currentDragItem: null,
			dndEventEmitter: new EventEmitter()
		};
	},


	childContextTypes: {
		//Common:
		currentDragItem: React.PropTypes.object,
		dndEvents: React.PropTypes.object,

		//For Draggable
		onDragStart: React.PropTypes.func,
		onDragEnd: React.PropTypes.func,
		onDrag: React.PropTypes.func,

		//For Droppable
		lastDragOver: React.PropTypes.object,
		onDragOver: React.PropTypes.func,
		onDrop: React.PropTypes.func
	},


	getChildContext () {
		let s = this.state;
		return {
			dndEvents: s.dndEventEmitter,
			currentDragItem: s.currentDragItem || null,
			lastDragOver: s.lastDragOver || null,

			onDragStart: this[onDragStart],
			onDragEnd: this[onDragEnd],
			onDrag: this[onDrag],

			onDragOver: this[onDragOver],
			onDrop: this[onDrop]
		};
	},


	getNewUniqueToken () {
		// This looks confusing, I know. This `token`
		// object is passed as a value to the DragTarget
		// accepts prop. Its also set to the type prop
		// of the Draggable. When a Draggable is over a
		// DropTarget, the DropTarget compares the
		// Draggable's type with its accepts list, if
		// it accepts it, the DropTarget will let the
		// Draggable drop onto it.
		//
		// The accepts function on this new instance of
		// an anonymouse object will test if the argument
		// passed to it is the exact same object as `token`.

		let token = { accepts: (t)=> t === token };

		return token;
	},


	getNewCombinationToken (...tokens) {
		return {
			accepts: (t)=> tokens.filter(x=> x === t || x.accepts(t)).length > 0
		};
	},


	[onDragStart] (item) {
		this.setState({
			currentDragItem: item,
			lastDragOver: null
		});
		emit(this, 'dragStart');
	},


	[onDragEnd] () {
		let lastOver = this.state.lastDragOver || {};
		let {target} = lastOver;
		let dropped = false;

		if (target) {
			dropped = target.handleDrop();
		}

		this.setState({
			currentDragItem: null
		});

		emit(this, 'dragEnd');
		return dropped;
	},


	[onDrag] (draggable, event, data) {
		emit(this, 'drag', data);
	},


	[onDragOver] (target, sender) {
		let last = this.state.lastDragOver || {};
		let lastTarget = last.target;

		if (!target && lastTarget && lastTarget !== sender) {
			return;
		}

		this.setState({
			lastDragOver: {
				source: this.state.currentDragItem,
				target: target
			}
		});
	},


	[onDrop] (target) {
		let drop = {
			source: this.state.currentDragItem,
			target: target
		};

		this.setState({
			lastDragOver: null
		});

		emit(this, 'drop', drop);

		if (this.onDrop) {
			this.onDrop(drop);
		}
	}
};
```

```
ERROR in ./src/main/js/common/dnd/Mixin.js
Module build failed: TypeError: /Users/jgrimes/Workspace/nti.web.mobile/src/main/js/common/dnd/Mixin.js: Cannot read property 'key' of null
    at add (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/scope/index.js:317:19)
    at add (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/scope/index.js:299:11)
    at Scope.generateUidIdentifierBasedOnNode (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/scope/index.js:322:5)
    at NodePath.exit (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/transformation/transformers/es6/properties.computed.js:93:25)
    at NodePath.call (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/path/context.js:55:28)
    at NodePath.visit (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/path/context.js:106:12)
    at TraversalContext.visitSingle (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/context.js:106:12)
    at TraversalContext.visit (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/context.js:118:19)
    at Function.traverse.node (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/index.js:66:17)
    at NodePath.visit (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/path/context.js:105:26)
    at TraversalContext.visitMultiple (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/context.js:86:16)
    at TraversalContext.visit (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/context.js:116:19)
    at Function.traverse.node (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/index.js:66:17)
    at NodePath.visit (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/path/context.js:105:26)
    at TraversalContext.visitSingle (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/context.js:106:12)
    at TraversalContext.visit (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/context.js:118:19)
    at Function.traverse.node (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/index.js:66:17)
    at traverse (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/index.js:49:14)
    at PluginPass.transform (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/transformation/plugin-pass.js:38:32)
    at File.transform (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/transformation/file/index.js:590:12)
    at /Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/transformation/pipeline.js:125:19
    at File.wrap (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/transformation/file/index.js:604:16)
    at Pipeline.transform (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/transformation/pipeline.js:122:17)
    at transpile (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-loader/index.js:12:22)
    at Object.module.exports (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-loader/index.js:69:12)
 @ ./src/main/js/common/dnd/index.js 17:13-31
```

**Comment by jsg2021** [comment]
Its worth noting that [our editor](//github.com/NextThought/react-editor-component) by itself builds and works just fine in 5.6.7... but when imported into my main project several of it's files trigger this error :/

As far as I can tell, the configuration is the same.

```
ERROR in ./~/react-editor-component/src/features/core.js
Module build failed: TypeError: /Users/jgrimes/Workspace/nti.web.mobile/node_modules/react-editor-component/src/features/core.js: Cannot read property 'key' of null
    at add (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/scope/index.js:317:19)
...
 @ ./~/react-editor-component/src/Editor.jsx 21:20-46

ERROR in ./~/react-editor-component/src/features/key-handlers.js
Module build failed: TypeError: /Users/jgrimes/Workspace/nti.web.mobile/node_modules/react-editor-component/src/features/key-handlers.js: Cannot read property 'key' of null
    at add (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/scope/index.js:317:19)
...
 @ ./~/react-editor-component/src/Editor.jsx 33:27-61

ERROR in ./~/react-editor-component/src/features/pointer-handlers.js
Module build failed: TypeError: /Users/jgrimes/Workspace/nti.web.mobile/node_modules/react-editor-component/src/features/pointer-handlers.js: Cannot read property 'key' of null
    at add (/Users/jgrimes/Workspace/nti.web.mobile/node_modules/babel-core/lib/babel/traversal/scope/index.js:317:19)
...
 @ ./~/react-editor-component/src/Editor.jsx 41:31-69
```

**Comment by sebmck** [comment]
Fixed as of 5.6.8, thanks!

**Comment by jsg2021** [comment]
no, thank you! babel is awesome :)
